source/megengine.js

```diff
@@ -3,30 +3,96 @@
         if (obj.__class__ && obj.__class__.__module__ === 'megengine.traced_module.traced_module' && obj.__class__.__name__ === 'TracedModule') {
-            return 'megengine.pickle';
+            megengine.modelType = 'megengine.pickle';
         }
-        return '';
+        return megengine.modelType;
     }
 
     open(context) {
         return context.metadata('megengine-metadata.json').then((metadata) => {
-            const obj = context.open('pkl');
-            return new megengine.Model(metadata, obj);
+            switch (megengine.modelType) {
+                case 'megengine.pickle': {
+                    const obj = context.open('pkl');
+                    return new megengine.Model(metadata, obj);
+                }
+                case 'megengine.flatbuffers': {
+                    return context.require('./megengine-schema').then(() => {
+                        megengine.schema_main = flatbuffers.get('megengine').mgb.serialization.fbs;
+                        megengine.schema_v2 = megengine.schema_main.v2;
+                        megengine.schema_param = megengine.schema_main.param;
+                        megengine.schema_dtype = megengine.schema_main;
+                        let model = null;
+                        const attachments = new Map();
+
+                        const stream = context.stream;
+                        stream.skip(4);
+                        try {
+                            const reader = flatbuffers.BinaryReader.open(stream);
+                            model = megengine.schema_v2.Model.create(reader);
+                        } catch (error) {
+                            const message = error && error.message ? error.message : error.toString();
+                            throw new megengine.Error('File format is not megengine.Model (' + message.replace(/\.$/, '') + ').');
+                        }
+                        try {
+                            const archive = zip.Archive.open(stream);
+                            if (archive) {
+                                for (const entry of archive.entries) {
+                                    attachments.set(entry[0], entry[1]);
+                                }
+                            }
+                        } catch (error) {
+                            // continue regardless of error
+                        }
+                        return new megengine.Model(metadata, model);
+                    });
+                }
+                default: {
+                    throw new megengine.Error("Unsupported megengine format '" + megengine.modelType + "'.");
+                }
+            }
         });
     }
 };
 
 megengine.Model = class {
 
     constructor(metadata, obj) {
-        this._format = 'MegEngine' + (obj.dump_info && obj.dump_info.version ? ' v' + obj.dump_info.version : '');
-        this._graphs = [ new megengine.Graph(metadata, obj) ];
+        switch (megengine.modelType) {
```

Comment:
Remove `megengine.modelType` and pass `modelType` via constructor.

Reply:
Since not every class needs to use megengine.modelType, for example, in Attribute, TensorType and Argument, but not Parameter and TensorShpae, will it be more flexible and clear to define it as a global variable?

Reply:
Some of the objects might expand lazily or multiple formats get loaded by the same code. The code for other formats consistently passes such information down the object hierarchy for classes that need it. The more basic question seems why does MegEngine create multiple different file formats that are very similar to formats that already exist?

Reply:
I checked with the official personnel that #606 was a pr proposed by the community personnel, not official. Therefore, it seems to be a coincidence that the file is defined as mge. In fact, it is not an object of any model format, but an ordinary pickled Python object, without any special features and description of the model structure. And as far as I know, internally, mge is only used as a c++ model format.

Reply:
done